artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederatedAbstract.java

```diff
@@ -104,9 +109,27 @@ public synchronized void createRemoteConsumer(FederatedConsumerKey key, Transfor
       if (started) {
          FederatedQueueConsumer remoteQueueConsumer = remoteQueueConsumers.get(key);
          if (remoteQueueConsumer == null) {
+            if (server.hasBrokerFederationPlugins()) {
+               try {
+                  server.callBrokerFederationPlugins(plugin -> plugin.beforeCreateFederatedQueueConsumer(key));
```

Comment:
extract interface, so that plugin does not expose concrete implementation, but simply api.